src/dave_data/datapool/hotmaps/hotmaps_request.py

```diff
@@ -0,0 +1,337 @@
+    if not isinstance(vector_path, str):
+        raise ValueError(f"Vector path '{vector_path}' is not a valid string.")
+
+    # Check if epsg is a valid integer
+    if not isinstance(epsg, int):
+        raise ValueError(f"EPSG code '{epsg}' is not a valid integer.")
+
+    with rasterio.open(clipped_raster_path) as src:
+        # Read the first band of the raster image
+        image = src.read(1)
+        # Create a mask for non-zero values
+        mask = image != 0
+
+        # Extracting the CRS from the raster
+        raster_crs = src.crs
+        raster_transform = src.transform
+
+        # Generate shapes (polygons) from the raster
+        results = (
+            {"properties": {"value": value}, "geometry": geometry}
+            for geometry, value in shapes(
+                image, mask=mask, transform=raster_transform
+            )
+        )
+
+        # Collect the features
+        features = list(results)
+
+        # Convert the extracted shapes to a GeoDataFrame
+        geometries = [shape(feature["geometry"]) for feature in features]
+        values = [feature["properties"]["value"] for feature in features]
+        vector = gpd.GeoDataFrame(
+            {"geometry": geometries, "value": values}, crs=raster_crs
+        )
+        # Reproject the vector to the EPSG of interest
+        vector_proj = vector.to_crs(epsg=epsg)
+
+        # Save the Geo-dataframe as a Geopackage
+        vector_proj.to_file(vector_path, driver="GPKG")
+        print(
+            f"Vector file '{vector_path}' corresponding to the georeferenced raster file is "
+            f"generated and added to the hotmaps_output folder."
+        )
+        return vector_path
+
+
+def hotmaps_request(layer_name, bbox, epsg):
+    # Validate layer_name
+    if not isinstance(layer_name, str):
+        raise ValueError("layer_name must be a string.")
+
+    # Validate bbox
+    validate_bbox(bbox)
+
+    # Validate epsg
+    validate_epsg(epsg)
+
+    try:
+        title = get_hotmaps_layer_info(layer_name)[0]
+        tag = get_hotmaps_layer_info(layer_name)[1]
+        key = get_hotmaps_layer_info(layer_name)[2]
+        category = get_hotmaps_layer_info(layer_name)[3]
+    except Exception as e:
+        print(e)
+        return
+
+    # Create the temp and output directory if it doesn't exist
+    output_directory = Path("hotmaps_output")
+    temp_directory = Path("temp_output")
+
+    output_directory.mkdir(parents=True, exist_ok=True)
+    temp_directory.mkdir(parents=True, exist_ok=True)
+
+    # Download HotMaps raster temporarily in the tep directory
+    download_hotmaps_raster(title, tag, key, category, temp_directory)
+
+    # Clip raster based on bounding box of interest
+    clipped_raster = clip_raster(temp_directory, title, output_directory, bbox)
+
+    # Convert clipped raster to vector
+    vector = raster_to_vector(output_directory, title, epsg)
+
+    # Remove temp directory and all its contents.
```

Comment:
Why keep the gpkg and tif file? I would say everything can be deleted when you return a geodataframe. 

If the files are kept then they need unique names.